sklearn/impute/tests/test_impute.py

```diff
@@ -28,6 +28,11 @@
 from sklearn.impute._base import _most_frequent
 
 
+def _assert_array_equal_and_same_dtype(x, y):
+    assert_array_equal(x, y)
```

Comment:
For comparing floats, it is a little safer to use `allclose`. What do you think about adding another helper for `allclose` and using that for comparing float arrays?

```python
def _assert_allclose_and_same_dtype(x, y):
    assert_allclose(x, y)
    assert x.dtype == y.dtype
```

Reply:
Done.